org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -14246,6 +14265,8 @@ private void proposeConstructor(
 						proposal.setReplaceRange(this.endPosition - this.offset, this.endPosition - this.offset);
 						proposal.setTokenRange(this.tokenStart - this.offset, this.tokenEnd - this.offset);
 						proposal.setRelevance(relevance);
+						proposal.setDeclarationTypeVariables(typeVariables);
+						proposal.setCompatibleProposal(isCompatibleType);
```

Comment:
Is there a reason why we don't call  `setDeclarationTypeVariables(..)` & `setCompatibleProposal(..)` for the default case (constructor with parameter) (in the else) ?

Reply:
Fixed.